File: 11th Grade/TP/h1/1.cpp
/*
#school - ТУЕС
#class - 11Б
#num - 22
#name - Михаил Михайлов Здравков
#task - Да се разработи програма, която изисква от потребителя да въведе две целочислени числа, x и y, където x < y. 
Да се намери сумата на всички нечетни числа в интервала [x,y].
*/

#include<iostream>
#include<math.h>
#include<stdlib.h>

using namespace std;

int main(){
    int x, y;

    int i, sum = 0;

    cout<< "Enter x: "<<endl;
    cin>> x;
    cout<<"\nEnter y > x: "<< endl;
    do{
       cin>> y;
    }while(y <= x);

    cout<<endl;

    for (i = x; i < y; i++){
        if ((i % 2)){
           sum += i;
        }
    }
    
    cout<< sum <<endl;
    
    return 0;
}

File: TP/h3/1.rb
#school - ТУЕС
#class - 11Б
#num - 15
#name - Добромир Иванов
#task - Да се създаде програма, която обработва CSV файл и изкарва на стандартния изход общия брой решени задачи на
#определените езици. 

#!/usr/bin/env ruby
#execute with either ruby filename.rb
#or chmod +x filename.rb and ./filename.rb
require "csv"

def parse_csv
	py = 0
	rb = 0
	sh = 0
	sum = 0
	CSV.foreach("2.csv") do |row|
		case row[5].downcase
			when "ruby" then rb += 1
			when "python" then py += 1
			when "shell" then sh += 1
		end
		sum += row[4].to_i
	end
	puts "ruby used " + rb.to_s + " times"
	puts "python used " + py.to_s + " times"
	puts "shell used " + sh.to_s + " times"
	puts "student class numbers sum: " + sum.to_s
end

parse_csv
